### PR TITLE
GitHub copilot chat -  @powerpages contribution

### PR DESCRIPTION
**Pull request by amitjoshi438**
This pull request primarily introduces a new chat participant feature to the Power Pages extension. The most significant changes include the addition of the chat participant in the `package.json` file, the creation of the `PowerPagesChatParticipant` class, and the inclusion of the chat participant in the extension's activation process.

image

image

Chat Participant Configuration:

* [`package.json`](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R104-R111): Added `github.copilot-chat` to `extensionDependencies` and introduced a new chat participant named `powerpages` in the `contributes` section. [[1]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R104-R111) [[2]](diffhunk://#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519R123-R130)

Chat Participant Implementation:

* [`src/client/extension.ts`](diffhunk://#diff-70fa1444b963acf260ba9443bca34c5f078ec0f23bbc3eb5cb5eaa00dc664baaR43): Imported `PowerPagesChatParticipant` and instantiated it during the extension's activation process. [[1]](diffhunk://#diff-70fa1444b963acf260ba9443bca34c5f078ec0f23bbc3eb5cb5eaa00dc664baaR43) [[2]](diffhunk://#diff-70fa1444b963acf260ba9443bca34c5f078ec0f23bbc3eb5cb5eaa00dc664baaR181-R183)
* [`src/common/chat-participants/ChatParticipantUtils.ts`](diffhunk://#diff-776fde87682985e6a43d5038599a6b92ad985e4a539979d54a5b87a6b5835dfaR1-R10): Created a utility function `createChatParticipant` to simplify the creation of chat participants.
* [`src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts`](diffhunk://#diff-885f58fd5bd7ee590ddae0f7854d056026aff9b576864e07f61887fb7586da29R1-R63): Introduced the `PowerPagesChatParticipant` class that handles chat requests and responses.
* [`src/common/chat-participants/powerpages/PowerPagesChatParticipantTypes.ts`](diffhunk://#diff-4f01d2df1e8f89d2ca7dc846477ced24118d0a76a97d02470a2721a2934508c6R1-R12): Defined the `IPowerPagesChatResult` interface to structure the chat result.

Chat Participant Assets:

* [`src/common/chat-participants/powerpages/assets/copilot.png`](diffhunk://#diff-2f146db89e86532aa9f408f9094cc039880d59b8dfd580bdcc41ac1fb8237661R1-R3): Added a new icon for the Power Pages chat participant.

### CODE REVIEW

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    private handler: vscode.ChatRequestHandler = async (
+        request: vscode.ChatRequest,
+        context: vscode.ChatContext,
+        stream: vscode.ChatResponseStream,
+        token: vscode.CancellationToken
+    ): Promise<IPowerPagesChatResult> => {
+        // Handle chat requests here
+
+        // TODO: Remove hardcoded values
+        const aibAuth = await intelligenceAPIAuthentication(this.telemetry, 'sessionID', '9ba620dc-4b37-430e-b779-2f9a7e7a52a6', true)
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:38]: lets not even checkin this hardcoded value  - we will miss this later - the quality needs to be on par with prod code here

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+        if(aibAuth.accessToken) {
+
+            // User is signed in
+            stream.markdown("Welcome to Power Pages! 🚀")
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:43]: localization missing

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+            //TODO: Add logic to handle the request
+            console.log(request)
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:46]: no console logs - leave a TODO command instead

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+            console.log(context)
+            console.log(token)
+
+        } else {
+            // User is not signed in
+            stream.markdown("Please sign in to use Power Pages")
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:52]: loc

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+        }
+
+        return {
+            metadata: {
+                command: 'powerpages'
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:57]: use a single enum to maintain all commands in single place

File: src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts [hunk 6]
```diff
@@ -0,0 +1,58 @@
+/*
```
Comment (tyaginidhi) [inline src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts:1]: for png above - i believe we should be using the new colorful logo

File: src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
```
Comment (tyaginidhi) [inline src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts:5]: nit: extra empty lines

File: src/common/chat-participants/vscode.proposed.languageModels.d.ts [hunk 8]
```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ */
+
```
Comment (tyaginidhi) [inline src/common/chat-participants/vscode.proposed.languageModels.d.ts:5]: nit: extra empty lines

**Review by tyaginidhi — CHANGES_REQUESTED** [review]

File: src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts (hunk 6 above)
Comment (amitjoshi438) [reply src/common/chat-participants/vscode.proposed.chatVariableResolver.d.ts:1]: I'll check with Manajit on this

File: src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+    private chatParticipant: vscode.ChatParticipant;
+    private telemetry: ITelemetry;
+
+    constructor(context: vscode.ExtensionContext, telemetry: ITelemetry | TelemetryReporter,) {
+
+        this.chatParticipant = createChatParticipant('powerpages', this.handler);
+
+        //TODO: Check the icon image
+        this.chatParticipant.iconPath = vscode.Uri.joinPath(context.extensionUri, 'src', 'common', 'chat-participants', 'powerpages', 'assets', 'copilot.png');
+
```
Comment (tyaginidhi) [inline src/common/chat-participants/powerpages/PowerPagesChatParticipant.ts:23]: nit: extra lines - make sure you have right extension for eslint installed

**Review by tyaginidhi — APPROVED** [review]

**Review by gshivi — APPROVED** [review]

**Pull request merged**